signal/micro/kernels/pcan.cc

```diff
@@ -0,0 +1,129 @@
+
+#include <stddef.h>
+#include <stdint.h>
+
+#include "signal/src/pcan_argc_fixed.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/flatbuffer_utils.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/memory_helpers.h"
+#include "tensorflow/lite/micro/micro_context.h"
+
+namespace tflite {
+namespace tflm_signal {
+namespace pcan {
+
+constexpr int kInputTensor = 0;
+constexpr int kNoiseEstimateTensor = 1;
+constexpr int kGainLutTensor = 2;
+constexpr int kOutputTensor = 0;
+
+// Indices into the init flexbuffer's vector.
+// The parameter's name is in the comment that follows.
+// Elements in the vectors are ordered alphabetically by parameter name.
+constexpr int kSnrShiftIndex = 0;  // 'snr_shift'
+
+struct TfLitePcanParams {
+  int snr_shift;
+};
+
+void* Init(TfLiteContext* context, const char* buffer, size_t length) {
+  const uint8_t* buffer_t = reinterpret_cast<const uint8_t*>(buffer);
```

Comment:
Can we just move the reinterpret_cast into the FlexbufferWrapper arguments on line 36?

Reply:
Done